internal/models/user.go

```diff
@@ -370,6 +396,22 @@ func (u *User) Authenticate(ctx context.Context, password string, decryptionKeys
 
 	compareErr := crypto.CompareHashAndPassword(ctx, hash, password)
 
+	if !strings.HasPrefix(hash, crypto.Argon2Prefix) {
+		// check if cost exceeds default cost or is too low
+		cost, err := bcrypt.Cost([]byte(hash))
+		if err != nil {
+			return compareErr == nil, false, err
+		}
+
+		if cost > bcrypt.DefaultCost || cost == bcrypt.MinCost {
```

Comment:
threshold decided for rehashing - let me know if we want to make this stricter / more lenient

Reply:
I think it is fine as is